azure-pipelines.yml

```diff
@@ -149,6 +155,97 @@ jobs:
+    env:
+      SPARK_HOME: $(Build.BinariesDirectory)\spark-2.3.2-bin-hadoop2.7
+      DOTNET_WORKER_DIR: $(Build.BinariesDirectory)\Microsoft.Spark.Worker\Microsoft.Spark.Worker-0.9.0
+
+  - task: DotNetCoreCLI@2
+    displayName: 'E2E tests for Spark 2.3.3 with oldest compatible worker (0.9.0)'
+    inputs:
+      command: test
+      projects: '**/Microsoft.Spark.E2ETest/*.csproj'
+      arguments: '--configuration $(buildConfiguration)'
+    env:
+      SPARK_HOME: $(Build.BinariesDirectory)\spark-2.3.3-bin-hadoop2.7
+      DOTNET_WORKER_DIR: $(Build.BinariesDirectory)\Microsoft.Spark.Worker\Microsoft.Spark.Worker-0.9.0
+
+  - task: DotNetCoreCLI@2
+    displayName: 'E2E tests for Spark 2.3.4 with oldest compatible worker (0.9.0)'
+    inputs:
+      command: test
+      projects: '**/Microsoft.Spark.E2ETest/*.csproj'
+      arguments: '--configuration $(buildConfiguration)'
+    env:
+      SPARK_HOME: $(Build.BinariesDirectory)\spark-2.3.4-bin-hadoop2.7
+      DOTNET_WORKER_DIR: $(Build.BinariesDirectory)\Microsoft.Spark.Worker\Microsoft.Spark.Worker-0.9.0
+
+  - task: DotNetCoreCLI@2
+    displayName: 'E2E tests for Spark 2.4.0 with oldest compatible worker (0.9.0)'
+    inputs:
+      command: test
+      projects: '**/Microsoft.Spark.E2ETest/*.csproj'
+      arguments: '--configuration $(buildConfiguration)'
+    env:
+      SPARK_HOME: $(Build.BinariesDirectory)\spark-2.4.0-bin-hadoop2.7
+      DOTNET_WORKER_DIR: $(Build.BinariesDirectory)\Microsoft.Spark.Worker\Microsoft.Spark.Worker-0.9.0
+
+  - task: DotNetCoreCLI@2
+    displayName: 'E2E tests for Spark 2.4.1 with oldest compatible worker (0.9.0)'
+    inputs:
+      command: test
+      projects: '**/Microsoft.Spark.E2ETest/*.csproj'
+      arguments: '--configuration $(buildConfiguration)'
+    env:
+      SPARK_HOME: $(Build.BinariesDirectory)\spark-2.4.1-bin-hadoop2.7
+      DOTNET_WORKER_DIR: $(Build.BinariesDirectory)\Microsoft.Spark.Worker\Microsoft.Spark.Worker-0.9.0
+  
+  - task: DotNetCoreCLI@2
+    displayName: 'E2E tests for Spark 2.4.3 with oldest compatible worker (0.9.0)'
+    inputs:
+      command: test
+      projects: '**/Microsoft.Spark*.E2ETest/*.csproj'
+      arguments: '--configuration $(buildConfiguration)'
+    env:
+      SPARK_HOME: $(Build.BinariesDirectory)\spark-2.4.3-bin-hadoop2.7
+      DOTNET_WORKER_DIR: $(Build.BinariesDirectory)\Microsoft.Spark.Worker\Microsoft.Spark.Worker-0.9.0
+
+  - task: DotNetCoreCLI@2
+    displayName: 'E2E tests for Spark 2.4.4 with oldest compatible worker (0.9.0)'
+    inputs:
+      command: test
+      projects: '**/Microsoft.Spark*.E2ETest/*.csproj'
+      arguments: '--configuration $(buildConfiguration)'
+    env:
+      SPARK_HOME: $(Build.BinariesDirectory)\spark-2.4.4-bin-hadoop2.7
+      DOTNET_WORKER_DIR: $(Build.BinariesDirectory)\Microsoft.Spark.Worker\Microsoft.Spark.Worker-0.9.0
+
+  - task: DotNetCoreCLI@2
+    displayName: 'E2E tests for Spark 2.4.5 with oldest compatible worker (0.9.0)'
```

Comment:
Can you verify 0.8.0 fails?

Reply:
Verified that it fails against 0.8.0

Reply:
Looks like it's failing because it cannot start the worker:
```
[2020-03-28T23:31:41.5990938Z] [fv-az213] [Debug] [ConfigurationService] Using the environment variable to construct .NET worker path: D:\a\1\b\Microsoft.Spark.Worker\Microsoft.Spark.Worker-{{oldestCompatibleWorker}}\Microsoft.Spark.Worker.exe.
[2020-03-28T23:31:42.0495263Z] [fv-az213] [Error] [JvmBridge] JVM method execution failed: Nonstatic method 'collectToPython' failed for class '19' when called with no arguments
[2020-03-28T23:31:42.0496295Z] [fv-az213] [Error] [JvmBridge] org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 1.0 failed 1 times, most recent failure: Lost task 0.0 in stage 1.0 (TID 1, localhost, executor driver): java.io.IOException: Cannot run program "D:\a\1\b\Microsoft.Spark.Worker\Microsoft.Spark.Worker-{{oldestCompatibleWorker}}\Microsoft.Spark.Worker.exe": CreateProcess error=2, The system cannot find the file specified
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1048)
```

Reply:
oops I see, ok let me fix that.

Reply:
I don't think it is possible to use variables in middle of another yaml property, they can only be used as anchor nodes and separate nodes.

Reply:
0.8.0 failing with the following error - Caused by: org.apache.spark.api.python.PythonException: System.MissingMethodException: Method not found: 'Microsoft.Spark.Sql.Row Microsoft.Spark.Sql.Row.op_Implicit(Microsoft.Spark.Sql.GenericRow)'.